File: src/main/kotlin/application/controller/dialogs/rule/isotropic/DeficientINTDialog.java
package application.controller.dialogs.rule.isotropic;

import application.controller.dialogs.rule.RuleWidget;
import application.model.rules.isotropic.rules.DeficientINT;

public class DeficientINTDialog extends RuleWidget {
    public DeficientINTDialog() {
        super();

        ruleFamily = new DeficientINT();
    }

    @Override
    public void updateRule(String rulestring) {
        ruleFamily.setRulestring(rulestring);
    }
}

File: src/main/kotlin/application/controller/dialogs/rule/isotropic/DeficientINTDialog.kt
package application.controller.dialogs.rule.isotropic

import application.controller.dialogs.rule.RuleWidget
import application.model.rules.isotropic.rules.DeficientINT

class DeficientINTDialog : RuleWidget() {
    override fun updateRule(rulestring: String?) {
        ruleFamily!!.rulestring = rulestring
    }

    init {
        ruleFamily = DeficientINT()
    }
}